waspc/data/Generator/templates/server/src/dbClient.js

```diff
@@ -12,6 +13,41 @@ const createDbClient = () => {
   const prismaClient = new Prisma.PrismaClient()
 
   {=# isAuthEnabled =}
+
+  prismaClient.$use(async (params, next) => {
+    // Ensure strong plaintext password. Must come before hashing middleware.
+    // Throws an EntityValidationError on the first validation that fails.
+    if (params.model === '{= userEntityUpper =}' && params.action === 'create') {
+        const data = params.args.data || {}
+
+        const defaultValidations = [
+          { name: 'email must be present', fn: data => !!data.email },
+          { name: 'password must be present', fn: data => !!data.password },
+          { name: 'password must be at least 8 characters', fn: data => data.password.length >= 8 },
+          { name: 'password must contain a number', fn: data => /\d/.test(data.password) },
+        ]
+
+        let validations = params.args._waspSkipDefaultValidations ? [] : defaultValidations
+        if (Array.isArray(params.args._waspCustomValidations)) {
+          validations = validations.concat(params.args._waspCustomValidations)
+        }
+
+        for (const validation of validations) {
+          if (!validation.fn(data)) {
+            throw new EntityValidationError('{= userEntityUpper =}', validation.name)
+          }
+        }
+
+        // Remove from downstream Prisma processing to avoid "Unknown arg" error
+        delete params.args._waspSkipDefaultValidations
```

Comment:
A question regarding naming these params -> you prefixed them with `_`, which makes a lot of sense, but I am guessing we will be having more of similar params in the future so it might be a good moment to discuss the prefix a bit.

One prefix I noticed used before is `$` -> I remember that being used in some JS libraries, although I am not sure which ones, to denote special params that are e.g. not stable or smth like that.

I am not saying that `$` is better than `_`, I really don't know yet, but just wondering what do you think about it and why `_`!

Reply:
Great question, I think the reason for me personally is 2 fold:
- In other languages, the _ denoted a convention for private. I think people then took that into JS: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_With_Private_Class_Features While this is not attempting to indicate private (nor does JS have that notion), it to me definitely signifies "off label" from actual Prisma object keys if you will, so it felt right from a visual shock perspective
- I think I have seen it used in similar contexts in other libraries with `_` (either at the beginning or end), but this is probably a pretty niche case of psuedo monkey patching where such conventions may not even exist?

Personally, anytime I see $ I think jQuery, but _ to many means Underscore.js too so 🤷🏻 😄 Happy to do a bit more digging before we decide on a convention moving forward if it helps. I don't have any preference here so long as it is apparent to users it is not built into Prisma and is Wasp specific. 👍🏻 Thanks!

Reply:
Same for me, `$` means jQuery and `_` is Underscore/lodash!

I don't have a preference, maybe I even like `_` better due to the reasons you explained.

Btw, if we wanted to be wild, we could even do smth like `args['{=waspSkipDefaultValidations=}`]` -> basically use strings for access which enables us to put any characters in the name -> but looking at it now it looks pretty ugly and irregular so I don't think it is a good idea.

Ok, if you want to do a bit of digging I will be interested to hear what you find, but in general if you decide to go with _ I am happy with that, it sounds good.